cirq-superstaq/cirq_superstaq/compiler_output.py

```diff
@@ -146,6 +147,54 @@ def __repr__(self) -> str:
             f"{self.pulse_lists!r})"
         )
 
+    def __repr_pretty__(self) -> str:
+        def _get_circuits_placeholder() -> str:
+            """Gets formatted placeholder for circuit object `repr`, since black can't parse it."""
+            if not self.has_multiple_circuits():
+                return "quantum_circuit"
+            if len(repr(self.circuits)) > 100:
+                # This forces black to multi-line format the placeholder for self.circuits.
+                placeholders = [
+                    f"quantum_circuit_{idx}" + "_" * 100 for idx in range(len(self.circuits))
+                ]
+                return f"[{', '.join(placeholders)}]"
+            return "quantum_circuits"
+
+        def _replace_placeholders(pretty_output: str) -> str:
+            """Replace placeholders for object `repr`s, since black can't parse them."""
+            if not self.has_multiple_circuits():
+                pretty_output = pretty_output.replace("quantum_circuit", repr(self.circuit))
```

Comment:
would it be possible/make sense to use the circuit diagrams instead of their reprs, assuming they're small enough?

Reply:
> would it be possible/make sense to use the circuit diagrams instead of their reprs, assuming they're small enough?

Yes, by the way.
![image](https://github.com/Infleqtion/client-superstaq/assets/18367737/5c9010a7-b899-47ef-b5f5-8f28f4c86e48)
Working on handling an edge case: empty circuits give empty pulse schedules which `black` complains about.

Reply:
Done. (Also moved pretty printing to separate files per chat with Rich.)